app/helpers/application_helper.rb

```diff
@@ -45,4 +45,10 @@ def url_params
   def render_turbo_stream_flash_messages
     turbo_stream.prepend "flashes", partial: "common/flash"
   end
+
+  def super_admin_acts_as_another_agent?
+    session[:rdv_solidarites_credentials].present? &&
+      session[:rdv_solidarites_credentials]["super_admin_id"].present? &&
+      session[:rdv_solidarites_credentials]["super_admin_id"].to_i != current_agent.id
```

Comment:
J'aimerai avoir confirmation qu'on estime donc que la session est un lieu sécurisé quoi qu'il arrive car pour pouvoir contourner ce système il faudrait pouvoir décrypter la session pour la modifier et donc avoir accès à la clé de chiffrement des sessions.

On peut aussi faire ca, mais ta version est peut être plus lisible en fait.
```ruby
super_admin_credentials = session.dig(:rdv_solidarites_credentials, "super_admin_id")
super_admin_credentials.present? && super_admin_credentials.to_i != current_agent.id
```

Reply:
> J'aimerai avoir confirmation qu'on estime donc que la session est un lieu sécurisé quoi qu'il arrive car pour pouvoir contourner ce système il faudrait pouvoir décrypter la session pour la modifier et donc avoir accès à la clé de chiffrement des sessions.

Oui c'est ça, tu penses que c'est pas assez safe ?

Reply:
j'aime bien ta version @Holist 
Pour la partie sécurisée je ne suis pas expert du sujet, j'ai eu cette crainte aussi et c'est effectivement ce qu'Amine m'a expliqué

Reply:
> > J'aimerai avoir confirmation qu'on estime donc que la session est un lieu sécurisé quoi qu'il arrive car pour pouvoir contourner ce système il faudrait pouvoir décrypter la session pour la modifier et donc avoir accès à la clé de chiffrement des sessions.
> 
> Oui c'est ça, tu penses que c'est pas assez safe ?

A priori je dirai que c'est relativement safe. On peut imaginer que si un attaquant à la clé de chiffrement des sessions il aura accès à d'autres éléments plus critiques que les cookies locaux du navigateur.
Je suis tombé sur ce post sur SO ou il conseille tout de même d'utiliser redis qui d'après l'auteur est plus secure et permet d'éviter ce problème pour les cookies les plus stratégiques comme ce qui concerne les droits d'accès... https://stackoverflow.com/questions/44220691/rails-what-are-the-consequences-of-a-leaked-secret-key-base

Reply:
Merci @Holist pour la suggestion.
Effectivement si notre clé d'encryption leak ce serait très problématique 😅 . 
J'envisageais peut-être de passer par redis en traitant #1912 mais j'évalue encore les pros et les cons